aws/resource_aws_codebuild_project.go

```diff
@@ -99,12 +99,18 @@ func resourceAwsCodeBuildProject() *schema.Resource {
 							ValidateFunc: validation.StringInSlice([]string{
 								codebuild.CacheTypeNoCache,
 								codebuild.CacheTypeS3,
+								codebuild.CacheTypeLocal,
 							}, false),
 						},
 						"location": {
 							Type:     schema.TypeString,
 							Optional: true,
 						},
+						"modes": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
One future enhancement that can be done here is supporting validation using the available [AWS Go SDK constants](https://docs.aws.amazon.com/sdk-for-go/api/service/codebuild/#pkg-constants):

```suggestion
							Elem:     &schema.Schema{
								Type: schema.TypeString,
								ValidateFunc: validation.StringInSlice([]string{
									codebuild.CacheModeLocalDockerLayerCache,
									codebuild.CacheModeLocalSourceCache,
									codebuild.CacheModeLocalCustomCache,
								},false),
							},
```